python/lsst/ts/m2gui/display/figure_constant.py

```diff
@@ -155,10 +155,14 @@ def _set_default_series(self, range_x_min, range_x_max, num_lines, text_legends)
         """
 
         chart = self.chart()
-        for idx in range(num_lines):
 
+        is_set_legend = len(text_legends) != 0
+        for idx in range(num_lines):
             series = QtCharts.QLineSeries()
 
+            if is_set_legend:
+                series.setName(text_legends[idx])
```

Comment:
This will raise IndexError if text_legends doesn't match. At least add that to Raises in docstring. Better would be probably to drop num_lines, and let the user create a series either providing an array in a variable called "lines", or "int" for a number of lines (which will be nameless). Of course, other solutions are possible as well.

Reply:
Improved the doc string in commit: [5d616a3](https://github.com/lsst-ts/ts_m2gui/pull/14/commits/5d616a3f591677c3b934045fd497728fe734ee07). Thanks!